OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -352,15 +352,17 @@ public static void SetupLatencyWidget(Widget parent, Session.Client c, OrderMana
 
 		public static void SetupProfileWidget(Widget parent, Session.Client c, OrderManager orderManager, WorldRenderer worldRenderer)
 		{
-			var visible = c != null && c.Bot == null;
 			var profile = parent.GetOrNull<ImageWidget>("PROFILE");
 			if (profile != null)
 			{
-				var imageName = (c != null && c.IsAdmin ? "admin-" : "player-")
-					+ (c.Fingerprint != null ? "registered" : "anonymous");
```

Comment:
This method can't decide on whether `c` might be null or not, sometimes it checks but other times it doesn't.

A preexisting issue, but might be nice to fix that whilst you're in here. At a cursory glance it appears all callers provide a non null client so I think removing the null checks is safe.

Reply:
Yes, indeed the whole thing wasn't that readable.